src/dmd/mtype.d

```diff
@@ -7353,3 +7353,49 @@ const(char)* toChars(ScopeRef sr) pure nothrow @nogc @safe
         return names[sr];
     }
 }
+
+/** Returns: `true` iff `t` has any non-`immutable` (mutable or const) indirections.
+ *
+ * In other words, if other references to `t` may be able to change it.
+ * Params:
+ *      t = type to check
+ * Returns:
+ *      true iff `t` contain non-`immutable` indirections.
+ */
+bool hasAliasing(scope Type t)
+{
+    t = t.toBasetype();
+    if (t.isImmutable)
+        return false;
+    else if (auto ts = t.isTypeStruct)
+    {
+        if (auto ti = ts.sym.isInstantiated())
+            if (ti.name.toString == "Rebindable" && // TODO: verify namespace std.typecons?
```

Comment:
no.

Reply:
But keep Rebindable specialization?

Reply:
You cannot make the compiler recognize a template by name.
this makes for unpredictable behavior.

Reply:
someone might used their own version of phobos with different Rebindable semantics in which case you still have to be correct.

Reply:
You can check the parent symbol and compare the module that the symbol came from, same as how built-ins are checked, although this is very discouraged because of what Stefan stated.

My idea about those built-ins is that D should define a `__builtin` module for that hardcoded modules, because this doesn't work if someone wants to create a new Phobos different from `std.` namespace and instead make an dummy module for that.

In this situation, this doesn't even work, so I would irradicate that check.

Reply:
Magic names need to be preceded with a double underscore.

Reply:
Replace `ti.name.toString == "Rebindable"` with `ti.name == Id.Rebindable`. The whole point of the Identifier system in dmd is to not need to do these string compares.

Reply:
> Magic names need to be preceded with a double underscore.

I'm not sure what you are exactly referring, but I'm referring to this https://github.com/dlang/dmd/blob/8f9012eaec8d2a09f2cd8acc548e3243c81408eb/src/dmd/builtin.d#L92

Reply:
Which files to I need to edit to register `Id.Rebindable`?

Reply:
> Which files to I need to edit to register `Id.Rebindable`?

`id.d` in the immutable `msgtable`.

Reply:
Added and used Id.Rebinable now.

Reply:
But this needs to be addressed https://github.com/dlang/dmd/pull/13040#discussion_r701964641 . I don't agree with hardcoding this.

Reply:
Why this was marked as resolved?

Reply:
There doesn't seem to be a consensus on supporting `Rebindable`. I'm not gonna waste any time until we've decide on whether to keep or not.

Reply:
> There doesn't seem to be a consensus on supporting `Rebindable`. I'm not gonna waste any time until we've decide on whether to keep or not.

You can at least check the `std.typecons` module instead of hardcoding it to every struct instance as I showed above, taking `dmd/builtin.d` as an example of it.

Reply:
I've remove any special handling of Rebindable.